src/reducers/versions.spec.tsx

```diff
@@ -124,6 +119,103 @@ describe(__filename, () => {
 
       expect(state.byAddonId[addonId]).toEqual(createVersionsMap(versions));
     });
+
+    it('sets the compare info to `null` on abortFetchDiff()', () => {
+      const versionsState = reducer(undefined, actions.abortFetchDiff());
+
+      expect(versionsState.compareInfo).toEqual(null);
+    });
+
+    it('resets the compare info on beginFetchDiff()', () => {
+      let versionsState = reducer(undefined, actions.abortFetchDiff());
+      versionsState = reducer(versionsState, actions.beginFetchDiff());
+
+      expect(versionsState.compareInfo).toEqual(undefined);
+    });
+
+    it('loads compare info', () => {
+      const addonId = 1;
+      const baseVersionId = 2;
+      const path = 'manifest.json';
+      const mimeType = 'mime/type';
+
+      const version = {
+        ...fakeVersionWithDiff,
+        file: {
+          ...fakeVersionWithDiff.file,
+          entries: {
+            ...fakeVersionWithDiff.file.entries,
+            [path]: {
+              ...fakeVersionWithDiff.file.entries[path],
+              mimetype: mimeType,
+            },
+          },
+          // eslint-disable-next-line @typescript-eslint/camelcase
+          selected_file: path,
+        },
+      };
+      const headVersionId = version.id;
+
+      let versionsState = reducer(
+        undefined,
+        actions.loadVersionInfo({ version }),
+      );
+      versionsState = reducer(
+        versionsState,
+        actions.loadDiff({
+          addonId,
+          baseVersionId,
+          headVersionId,
+          version,
+        }),
+      );
+
+      expect(versionsState.compareInfo).toEqual({
+        diffs: createInternalDiffs({
+          baseVersionId,
+          headVersionId,
+          version,
+        }),
+        mimeType,
+      });
+    });
+
+    it('logs a debug message when entry is missing on loadDiff()', () => {
+      const addonId = 1;
+      const baseVersionId = 2;
+      const path = 'some/other/file.js';
+      const _log = getFakeLogger();
+
+      const version = {
+        ...fakeVersionWithDiff,
+        file: {
+          ...fakeVersionWithDiff.file,
+          // eslint-disable-next-line @typescript-eslint/camelcase
+          selected_file: path,
+        },
+      };
+      const headVersionId = version.id;
+
+      const previousState = reducer(
+        undefined,
+        actions.loadVersionInfo({ version }),
+      );
+      // TS looks confused about the third optional argument.
+      // @ts-ignore
```

Comment:
It looks like the Reducer type has the last word: https://github.com/reduxjs/redux/blob/v4.0.1/index.d.ts#L58-L61

Huh. I guess technically our reducer shouldn't accept options. Ignoring it is fine for now but we'll want to use testing patterns like this in the future so it would be nice to have a solution.

We could work around it by introducing a factory pattern:

```js
const reducer = createReducer({ _log: getFakeLogger() });
...
const versionsState = reducer(...);
```